bundles/binding/org.openhab.binding.weather/src/main/java/org/openhab/binding/weather/internal/bus/WeatherBinding.java

```diff
@@ -71,15 +71,22 @@ public void deactivate() {
      */
     @Override
     public void updated(Dictionary<String, ?> config) throws ConfigurationException {
-        if (config != null) {
-            context.getJobScheduler().stop();
+        if (config == null) {
+            logger.warn("Unable to find any configuration settings for weather binding. Check openhab.cfg.");
+            throw new ConfigurationException("weather", "Unable to find any configuration settings for weather binding. Check openhab.cfg.");
+        }
```

Comment:
Are you certain that there aren't expected situations when `updated` is called with a `null` config?  A thorough exercise under OH 1.8.3 and OH 2 might reveal if this is correct; perhaps you've already done that?